content/en/docs/ops/diagnostic-tools/istioctl-analyze/test.sh

```diff
@@ -0,0 +1,115 @@
+set -u  # Unset is an error
+# There is no need to echo, output appears in ops_diagnostic-tools_istioctl-analyze_test_debug.txt
+set -o pipefail
+
+# This script doesn't need a control plane initially and will install Istio when needed
+# @setup profile=none
+
+# The test harness labels the default namespace.  Remove that label
+# so the output matches the expect output on a fresh K8s cluster.
+kubectl label namespace default istio-injection- || true
+
+echo '*** istioctl-analyze step 1 ***'
+_verify_contains snip_analyze_all_namespaces "$snip_analyze_all_namespace_sample_response"
+
+echo '*** istioctl-analyze step 2 ***'
+snip_fix_default_namespace
+_verify_same snip_try_with_fixed_namespace "$snip_try_with_fixed_namespace_out"
+
+echo '*** istioctl-analyze step 3 ***'
+_verify_contains snip_analyze_sample_destrule "$snip_analyze_sample_destrule_out"
+
+# TODO It is unclear if this should ALWAYS fail, NEVER fail, or sometimes fail.
+# There are multiple DestinationRules, some are valid for the VirtualService, some lack subsets 
+echo '*** istioctl-analyze step ***'
+snip_analyze_networking_directory || true
+
+echo '*** istioctl-analyze step 5 ***'
+snip_analyze_all_networking_yaml
+
+echo '*** istioctl-analyze step 6 ***'
+snip_analyze_all_networking_yaml_no_kube
+
+echo '*** istioctl-analyze step 7 ***'
+istioctl analyze --help
+
+echo '*** istioctl-analyze step 8 ***'
+snip_install_with_custom_config_analysis
+_wait_for_deployment istio-system istiod
+
+echo '*** istioctl-analyze step 9 ***'
+set +e  # Don't exit on failure
+kubectl apply -f - <<EOF
+apiVersion: networking.istio.io/v1alpha3
+kind: VirtualService
+metadata:
+  name: ratings
+  namespace: default
+spec:
+  gateways:
+  - bogus-gateway
+  hosts:
+  - ratings
+  http:
+  - route:
+    - destination:
+        host: ratings
+EOF
+set -e  # Exit on failure
+
+echo '*** istioctl-analyze step 10 ***'
+get_ratings_virtual_service() {
+kubectl get vs ratings -o yaml
+}
+_verify_elided get_ratings_virtual_service "$snip_vs_yaml_with_status"
+
+echo '*** istioctl-analyze step 11 ***'
+kubectl create ns frod
+_verify_failure analyze_k_frod
+# TODO Replace the above with _verify_failure_same when it is available
+# _verify_same analyze_k_frod "$snip_analyze_k_frod_out"
+
+echo '*** istioctl-analyze step 12 ***'
+_verify_same snip_analyze_suppress0102 "$snip_analyze_suppress0102_out"
+
+echo '*** istioctl-analyze step 13 ***'
+# TODO This will return errors on non-Istio namespaces, e.g. ibm-cert-store and kube-node-lease
+# It is hard to know what to supply to fix that problem for testing.
+snip_analyze_suppress_frod_0107_baz || true
+
+echo '*** istioctl-analyze step 14 ***'
+kubectl create deployment my-deployment --image=docker.io/kennethreitz/httpbin
+snip_annotate_for_deployment_suppression
+
+echo '*** istioctl-analyze step 15 ***'
+kubectl annotate deployment my-deployment galley.istio.io/analyze-suppress-
+snip_annotate_for_deployment_suppression_107
+
+# @cleanup
```

Comment:
The cleanup commands needs to ignore errors, since cleanup may be called early (before some of the things are even created) if the test fails. If you look at other tests, you'll notice that every one has `set +e` as the first line. Maybe the framework should inject that automatically (like it does some of the include files?)

```suggestion
# @cleanup
set +e # ignore cleanup errors
```